src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,51 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
```

Comment:
Parameterize `MachineService` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created. For example, this code should not compile: `public class StringService implements MachineService<String> { }`